Allow and demonstrate NNG usage with CMake FetchContent

dksmiffs opened the pull request:
@gdamore, this PR breaks two CONTRIBUTING rules, but for these reasons:

1. I couldn't squash to one commit, because the second commit needs the SHA1 of the first.
2. If you accept this change, a follow-up change will be required to the new demo's `CMakeLists.txt`, to refer to your repo instead of my fork.

I hope you understand this awkwardness was necessary because the change I'm suggesting isn't in your repo yet, and yet my demo's `CMakeLists.txt` requires SHA1 access to the repo itself with the single line suggested change to the top-level `CMakeLists.txt`.

I promise I will respond to review feedback, and not leave this baby on your doorstep.

gdamore commented:
So lets fix this to refer to the main NNG repo instead.  Let me see if I can take this work and adjust it to work that way.

dksmiffs commented:
@gdamore, after more thought, I don't like the fact that I've overloaded the meaning of `nng::nng`. The `nng::nng` that I introduced in the top level `CMakeLists.txt` file is a CMake library alias.  That's not the same thing as the value of the `NNG_LIBRARY` variable you have set to `nng::nng` in `cmake/nng-config.cmake.in`.  Maybe those overloaded names wouldn't conflict, but I don't have enough experience with CMake config packages to say for sure.

I would feel much more confident if we renamed the library alias I introduced to something unique like `nng::nnglib`.  If you'd be okay with this tweak, I'd be happy to make the change in this pull request.

dksmiffs commented:
@gdamore After still more thought, I recommend that you close this pull request without merging it into mainline. Your current way works, and does NOT require elevated privileges. Here are widely used CMake build steps that already meet my original intent for writing this pull request:
```bash
cmake -G Ninja -B ./build -S . -DCMAKE_INSTALL_PREFIX=/some/writeable/directory
cmake --build ./build
cd build
ctest
cmake --build . --target install
```
This is the way I'll be using NNG moving forward. Thanks for your hard work on NNG!

gdamore commented:
Closed at request of submitter.
